components/brave_vpn/brave_vpn_service.cc

```diff
@@ -662,23 +688,52 @@ void BraveVpnService::OnPrepareCredentialsPresentation(
     SetPurchasedState(env, PurchasedState::NOT_PURCHASED);
     return;
   }
+
+  // Early return when it's already expired.
+  if (time < base::Time::Now()) {
+    SetPurchasedState(env, PurchasedState::EXPIRED);
+    return;
+  }
+
   if (GetCurrentEnvironment() != env) {
     // Change environment because we have successfully authorized with new one.
     SetCurrentEnvironment(env);
   }
 
-  skus_credential_ = credential;
+  api_request_.GetSubscriberCredentialV12(
+      base::BindOnce(&BraveVpnService::OnGetSubscriberCredentialV12,
+                     base::Unretained(this), time),
+      credential, GetBraveVPNPaymentsEnv(GetCurrentEnvironment()));
+}
 
+void BraveVpnService::OnGetSubscriberCredentialV12(
+    const base::Time& expiration_time,
+    const std::string& subscriber_credential,
+    bool success) {
+  if (!success) {
+    VLOG(2) << __func__ << " : failed to get subscriber credential";
 #if BUILDFLAG(IS_ANDROID)
-  SetPurchasedState(env, PurchasedState::PURCHASED);
+    SetPurchasedState(GetCurrentEnvironment(), PurchasedState::NOT_PURCHASED);
 #else
-  GetBraveVPNConnectionAPI()->SetSkusCredential(skus_credential_);
+    if (subscriber_credential == kTokenNoLongerValid) {
```

Comment:
@spylogsster Curious `kTokenNoLongerValid` is only effective on desktop?

Reply:
I do not know, maybe @deeppandya could check guardian response on Android?

Reply:
Code is consistent with before at least - so if it was used on mobile, it's continuing to be used 👍 And for expired case, we default to showing in-app-purchase again on mobile